src/contracts/atlas/ExecutionEnvironment.sol

```diff
@@ -76,14 +67,24 @@ contract ExecutionEnvironment is Base {
         _;
     }
 
+    modifier contributeSurplus() {
+        _;
+        {
+        uint256 balance = address(this).balance;
+        if (balance > 0) {
+            IEscrow(atlas).contribute{value: balance}();
+        }
+        }
+    }
+
     //////////////////////////////////
     ///    CORE CALL FUNCTIONS     ///
     //////////////////////////////////
     function preOpsWrapper(UserOperation calldata userOp)
         external
         validUser(userOp)
         onlyAtlasEnvironment(ExecutionPhase.PreOps, _ENVIRONMENT_DEPTH)
-        contributeSurplus(Party.DApp)
+        contributeSurplus
         returns (bytes memory)
```

Comment:
Should this be `payable`? It's receiving value from the EE call.

Reply:
Yes, good catch, we should add `payable` to any functions that could have ETH flowing through. All it really does under the hood is remove the `require(msg.value == 0)` check that the solidity compiler adds in non-payable functions

Will add payable wherever contributeSurplus is used